### PR TITLE
Generate malleated p2pkh vectors

### PR DESCRIPTION
**Pull request by Eunovo**
- Add vectors for malleated p2pkh scripts. Test:
  - OP_0 OP_DROP <scriptsig>
  - OP_1 OP_IF <real_scriptsig> OP_ELSE <fake_scriptsig> OP_ENDIF

### CODE REVIEW

**Comment by S3RK** [comment]
If we decided to remove `generate_all_inputs_test` could you also then remove corresponding parts from it? As it already has malted p2pkh inputs there.

cc @josibake

**Comment by josibake** [comment]
> could you also then remove corresponding parts from it

~Personally, I find it easier to keep the `generate_all_inputs` test as it is right now as we split the tests out, because we can easily verify at the end that we've covered all of the cases we want from the `generate_all_inputs` test and addressed the TODOs from the test.~

Actually, I take that back :sweat_smile: , let's remove the code from the `generate_all_inputs` test as we go.

**Comment by josibake** [comment]
> * Add vectors for p2sh p2ms

I don't see this currently on the test, and probably better to add as it's own PR? Makes it easier to review and reason about each test case if its only testing one thing.

@Eunovo if you update this to remove the dead code from the `generate_all_inputs` test and update the description, I think this looks good to go.

**Comment by Eunovo** [comment]
> I don't see this currently on the test, and probably better to add as it's own PR? Makes it easier to review and reason about each test case if its only testing one thing.

@josibake Yeah, I forgot to push some commits. I've moved them to another branch and I'll open a PR for them

**Comment by Eunovo** [comment]
> @Eunovo if you update this to remove the dead code from the `generate_all_inputs` test and update the description, I think this looks good to go.

Done

File: generate-test-vector.py [hunk 1]
```diff
@@ -1241,6 +1227,99 @@ def generate_taproot_with_nums_point_test():
...
+
+    # Malleated p2pkh
+    # Test OP_O OP_DROP <normal script sig>
+    add_input(inputs, input_priv_keys, input_pub_keys, lambda pub, priv: "0075" + get_p2pkh_scriptsig(pub, priv))
+    # Test make dummy key look like private key
+    # Ensures that reference implementation p2pkh can extract correct key
+    # 01 OP_IF <real_script> ELSE <fake_script> ENDIF
+    fake_priv, fake_pub = get_key_pair(0, seed=bytes.fromhex("faecaa"))
+    fake_script = get_p2pkh_scriptsig(fake_pub, fake_priv)
+    add_input(inputs, input_priv_keys, input_pub_keys, lambda pub, priv: "0163" + get_p2pkh_scriptsig(pub, priv) + "67" + fake_script + "68")
```
Comment (S3RK) [inline generate-test-vector.py]: `OP_1` is 0x51 not 0x01

Comment (S3RK) [reply generate-test-vector.py]: I use btcdeb to compile and execute scripts

**Comment by S3RK** [comment]
one problem with maleated script, the rest is LGTM

**Comment by Eunovo** [comment]
@S3RK  I've addressed the problem with the script.

**Comment by josibake** [comment]
ACK https://github.com/S3RK/bip-352/commit/93906fcbedf48b8a4b247e4ad4865953386d7976

**Pull request merged**